src/pages/settings/profile/index.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useState } from 'react';
+import { Auth } from 'aws-amplify';
+import _ from 'lodash';
+import {
+  Form, Input, Empty, PageHeader, Card, Row, Col, Button, Space,
+} from 'antd';
+import { useRouter } from 'next/router';
+import FeedbackButton from '../../../components/FeedbackButton';
+import endUserMessages from '../../../utils/endUserMessages';
+import pushNotificationMessage from '../../../utils/pushNotificationMessage';
+
+const ProfileSettings = () => {
+  const router = useRouter();
+
+  const [user, setUser] = useState();
+  const [oldPasswordError, setOldPasswordError] = useState(null);
+  const [newPasswordError, setNewPasswordError] = useState(null);
+  const [emailError, setEmailError] = useState(null);
+
+  const initialState = {
+    changedPasswordAttributes: {},
+    changedUserAttributes: {},
+  };
+  const [newAttributes, setNewAttributes] = useState(initialState);
+  const { changedPasswordAttributes, changedUserAttributes } = newAttributes;
+
+  const setChanges = (object) => {
+    const newChanges = _.cloneDeep(newAttributes);
+    _.merge(newChanges, object);
+    setNewAttributes(newChanges);
+  };
+
+  const currentUser = () => Auth.currentAuthenticatedUser()
+    .then((userData) => setUser(userData))
+    .catch((e) => console.log('error during getuser', e));
+
+  useEffect(() => {
+    currentUser();
+  }, []);
+
+  const updateDetails = async () => {
+    const { name, email } = changedUserAttributes;
+    const { oldPassword, newPassword, confirmNewPassword } = changedPasswordAttributes;
+
+    const invalidPasswordErrors = ['InvalidPasswordException', 'InvalidParameterException', 'NotAuthorizedException'];
+    if (name || email) {
+      setEmailError(false);
+      await Auth.updateUserAttributes(user, changedUserAttributes)
+        .then((response) => pushNotificationMessage('success', endUserMessages.ACCOUNT_DETAILS_UPDATED, 3))
+        .catch((e) => setEmailError(true));
+    }
+    if (oldPassword || newPassword || confirmNewPassword) {
+      setOldPasswordError(false);
+      setNewPasswordError(false);
+      const decimal = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,}$/;
```

Comment:
This regex should be renamed. And probably also add a comment reminding people that it should be kept up to date with the Cognito validation policy.